### PR TITLE
Integrate Ribo-TISH

### PR DESCRIPTION
**Pull request by pinin4fjords**
This PR wires in RiboTISH, and adds associated config, documentation etc. I also updated a bunch of modules, which is what makes the number of changed files a bit scary.

This is the first ribo-seq specific functionality, so there's also the wiring from the sample sheet to separate riboseq from rnaseq from TI-seq.

There's still some work to do on wider documentation etc, will do in a separate PR before release.

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/riboseq/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/riboseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: assets/samplesheet.csv
Comment (maxulysse) [inline assets/samplesheet.csv:1]: this has became highly detailed

File: conf/modules.config [hunk 1]
```diff
@@ -353,34 +363,68 @@ if (params.remove_ribo_rna) {
...
+        }
+
+        withName: 'NFCORE_RIBOSEQ:RIBOSEQ:.*:BAM_SORT_STATS_SAMTOOLS_GENOME:SAMTOOLS_SORT' {
+            ext.prefix = { "${meta.id}.genome.sorted" }
+            publishDir = [
+                path: { "${params.outdir}/alignment/${params.aligner}/sorted" },
                 mode: params.publish_dir_mode,
                 pattern: "*.bam",
-                saveAs: { ( ['star_salmon','hisat2'].contains(params.aligner) &&
+                saveAs: { ( ['star','hisat2'].contains(params.aligner) &&
```
Comment (maxulysse) [inline conf/modules.config:391]: why hisat2 if star is the only possible aligner?

**Review by maxulysse — COMMENTED** [review]
Looking good, just a bit confused with the aligner params

File: assets/samplesheet.csv
Comment (pinin4fjords) [reply assets/samplesheet.csv:1]: I'll strip the extraneous columns - thanks!

File: conf/modules.config (hunk 1 above)
Comment (pinin4fjords) [reply conf/modules.config:391]: Thanks!

File: assets/samplesheet.csv
Comment (maxulysse) [reply assets/samplesheet.csv:1]: Can you remove the quotes as well?

Comment (pinin4fjords) [reply assets/samplesheet.csv:1]: Why? Perfectly valid CSV

File: nextflow.config [hunk 2]
```diff
@@ -74,7 +74,7 @@ params {
     kallisto_quant_fraglen_sd  = 200
     save_merged_fastq          = false
     save_unaligned             = false
-    save_align_intermeds       = false
+    save_align_intermeds       = true
```
Comment (maxulysse) [inline nextflow.config:77]: Is there a need for this as default?

**Review by maxulysse — APPROVED** [review]
Looks good to me

File: nextflow.config (hunk 2 above)
Comment (pinin4fjords) [reply nextflow.config:77]: Yeah, I think the alignment files are more of a necessary output for riboseq, while in rnaseq they're only a stop on a path to quantification. 

I might reverse it later, but for now it was intentional.

**Comment by pinin4fjords** [comment]
Thanks for the review @maxulysse

**Pull request merged**